posthog/version_requirement.py

```diff
@@ -0,0 +1,68 @@
+from semantic_version.base import SimpleSpec, Version
+
+
+class VersionRequirement:
+    accepted_services = ("clickhouse", "postgres")
+
+    def __init__(self, service, version_range):
+        if service not in self.accepted_services:
+            services_str = ", ".join(self.accepted_services)
+            raise Exception(
+                f"service {service} cannot be used to specify a version requirement. service should be one of {services_str}"
+            )
+
+        self.service = service
+
+        try:
+            self.version_range = SimpleSpec(version_range)
+        except:
+            raise Exception(
+                f"version_range is invalid. See the Docs for SimpleSpec: https://pypi.org/project/semantic-version/"
+            )
+
+    def is_service_in_accepted_version(self):
+        service_version = self.get_service_version()
+        return [service_version in self.version_range, service_version]
+
+    def get_service_version(self):
+        if self.service == "postgres":
+            return self.get_postgres_version()
+
+        if self.service == "clickhouse":
+            return self.get_clickhouse_version()
+
+    def get_postgres_version(self):
+        from django.db import connection
+
+        with connection.cursor() as cursor:
+            cursor.execute("SHOW server_version")
```

Comment:
Did we verify pgbouncer doesn't intercept and bypass this query returning some custom output? I don't expect it but still better to verify it

Reply:
Just checked on Cloud 👍 

good point though